gr-fec/python/fec/polar/channel_construction_awgn.py

```diff
@@ -198,7 +198,7 @@ def upper_convolve(tpm, mu):
     idx = -1
     for i in range(mu):
         idx += 1
-        q[0, idx] = (tpm[0 / i] ** 2 + tpm[1, i] ** 2, 2)
+        q[0, idx] = (tpm[0, i] ** 2 + tpm[1, i] ** 2, 2)
```

Comment:
This might look like a simple change but it is part of this whole algorithm.

I wrote this code 7 years ago based on the paper "How to construct polar codes". I tried to validate these results by comparing them to the implementation I had available from Ido Tal. Everything looked fine back then.

For small configurations, e.g. 32 or 64, this should produce the same reliability order as other implementations, e.g. the Bhattacharyya Bounds method.

This needs more investigation. Maybe the referenced papers help.